src/NUnitEngine/nunit.engine.api/RuntimeType.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace NUnit.Engine
+{
+    /// <summary>
+    /// Enumeration identifying a common language
+    /// runtime implementation.
+    /// </summary>
+    public enum RuntimeType
+    {
+        /// <summary>Any supported runtime framework</summary>
+        Any,
+        /// <summary>Microsoft .NET Framework</summary>
+        Net,
+        /// <summary>Microsoft .NET Compact Framework</summary>
+        NetCF,
+        /// <summary>Microsoft Shared Source CLI</summary>
+        SSCLI,
+        /// <summary>Mono</summary>
+        Mono,
+        /// <summary>Silverlight</summary>
+        Silverlight,
+        /// <summary>MonoTouch</summary>
+        MonoTouch
```

Comment:
There are fields here that we don't use, of course, but they have been around  for a while. I think we should eventually merge the other platforms into this list, but the engine doesn't do anything with them right now.

In order to expose IAvailableRuntimes, I needed an interface for a runtime framework. I created IRuntimeFramework for that purpose. If memory serves, I asked you offline if you thought we should use the existing enumeration or a text field for this purpose and you suggested using the enum. Did we miscommunicate?

Basically, __something__ is needed in IRuntimeFramework to represent the particular type of runtime we are dealing with. It doesn't have to be the existing internal enum. It could be a new enum or it can be a string. Microsoft uses a string to represent the runtime in it's TargetFrameworkAttribute and we already use strings in lots of places in order to give us flexibility (at the cost of type-safety, of course).

I'm open to replacing this and even to changing the internal enum as appropriate, but we need something to distinguish the runtime in question beyond the simple display name. Eventually, this value will trigger the engine to do different things to run the tests depending on the runtime.